### PR TITLE
fix to allow single sided borders

### PR DESCRIPTION
**Pull request by seanwoodward**

### CODE REVIEW

File: IBAnimatable/BorderDesignable.swift [hunk 1]
```diff
@@ -25,16 +25,23 @@ public protocol BorderDesignable {
...
+    
     guard let unwrappedBorderColor = borderColor else {
+      clearBorders()
       return
     }
     
     if borderWidth.isNaN || borderWidth <= 0 {
+      clearBorders()
       return
     }
```
Comment (tbaranes) [inline IBAnimatable/BorderDesignable.swift]: Couldn't we merge that `if` in the `guard`?

**Comment by tbaranes** [comment]
Thanks @seanwoodward, nice catch :+1:

File: IBAnimatable/BorderDesignable.swift [hunk 2]
```diff
@@ -25,16 +25,23 @@ public protocol BorderDesignable {
...
+        .forEach { $0.removeFromSuperlayer() }      
+    }
+    
     guard let unwrappedBorderColor = borderColor else {
+      clearBorders()
       return
     }
     
     if borderWidth.isNaN || borderWidth <= 0 {
+      clearBorders()
```
Comment (tbaranes) [inline IBAnimatable/BorderDesignable.swift]: Couldn't we merge that if in the guard?
 Add a line note

File: IBAnimatable/BorderDesignable.swift (hunk 1 above)
Comment (tbaranes) [inline IBAnimatable/BorderDesignable.swift]: Couldn't we merge that if in the guard?

Comment (tbaranes) [inline IBAnimatable/BorderDesignable.swift:41]: Since the `guard` and the `if` are doing the thing, couldn't we merge them?

**Comment by tbaranes** [comment]
Nice catch :+1: 
I made a minor feedback just to improve that implementation, but looks good!

I'm gonna let @JakeLin make the final review :dancers:

**Comment by JakeLin** [comment]
Good one, :+1:

agree with @tbaranes ,  it is better to group them within one. Can we optimise that in this PR as well? Thanks.

**Comment by JakeLin** [comment]
@seanwoodward thanks again for your PR, do you have time to make the last change according to @tbaranes 's comment? Or we can merge this PR then make the change after. We are about to release 2.2, 😉 thanks.

**Comment by tbaranes** [comment]
I think we can merge it. We can handle that comment, or even do it directly in master since it's not a big change.

**Comment by JakeLin** [comment]
@seanwoodward thanks again for your PR. @tbaranes working on release 2.2 now.

**Pull request merged**